File: frontend/types/portfolio/project-media.ts
import type {
  ProjectMediaType,
} from "@/types/portfolio/common"

export type ProjectMediaRead = {
  id: string
  project_id: string

  media_type: ProjectMediaType

  url: string
  thumbnail_url: string | null

  alt_text: string | null
  caption: string | null

  provider: string | null
  provider_asset_id: string | null

  mime_type: string | null

  width: number | null
  height: number | null

  file_size_bytes: number | null
  duration_seconds: number | null

  is_primary: boolean
  sort_order: number

  created_at: string
  updated_at: string
  deleted_at: string | null
}

export type ProjectMediaSummary = {
  id: string
  project_id: string

  media_type: ProjectMediaType

  url: string
  thumbnail_url: string | null

  alt_text: string | null
  caption: string | null

  width: number | null
  height: number | null

  duration_seconds: number | null

  is_primary: boolean
  sort_order: number
}

File: mobile/shared/src/commonMain/kotlin/com/akahalu/portfolio/domain/portfolio/model/ProjectMedia.kt
package com.akahalu.portfolio.domain.portfolio.model

enum class ProjectMediaType {
    IMAGE,
    VIDEO,
    DOCUMENT,
    DEMO,
    OTHER,
}

data class ProjectMedia(
    val id: String,
    val projectId: String,
    val mediaType: ProjectMediaType,
    val url: String,
    val thumbnailUrl: String?,
    val altText: String?,
    val caption: String?,
    val provider: String?,
    val providerAssetId: String?,
    val mimeType: String?,
    val width: Int?,
    val height: Int?,
    val fileSizeBytes: Long?,
    val durationSeconds: Int?,
    val isPrimary: Boolean,
    val sortOrder: Int,
)